Sources/TensorFlow/Operators/Math.swift

```diff
@@ -2436,6 +2436,23 @@ internal extension Tensor where Scalar: TensorFlowFloatingPoint {
 // TODO: Consider making the return type be generic over `FloatingPoint` types
 // so that `self`'s scalar type can be any `Numeric` type.
 public extension Tensor where Scalar: TensorFlowFloatingPoint {
+
+    /// Helper function that assess if `axis` is in the range `[-rank, rank)`, where `rank` is the rank of
+    /// the provided tensors.
+    @usableFromInline
+    internal func isAxisInRange(_ axis: Int) -> Bool {
+        return axis >= -rank && axis < rank
+    }
+
+    @usableFromInline
+    internal func areAxesInRange(_ axes: [Int]) -> Bool {
+        return !axes.contains(where: { !isAxisInRange($0) })
+    }
+
+    @usableFromInline
+    internal func areAxesInRange(_ axes: Tensor<Int32>) -> Bool {
+        return !axes.scalars.contains(where: { !isAxisInRange(Int($0)) })
```

Comment:
Reuse code:
```suggestion
        return areAxesInRange(axes.scalars)
```

Reply:
I don't think it will work as axes.scalars will be of type `Int32` while the input to the above function is `Int`.

Reply:
I see, good point. In that case, we should make `isAxisInRange` and `areAxesInRange` generic to support both `Int` and `Int32`.

Could you try the following:
```swift
extension Tensor {
    func isAxisInRange<T: BinaryInteger>(_ axis: T) -> Bool {
        let axis = Int(axis)
        return axis >= -rank && axis < rank
    }

    func areAxesInRange<T: BinaryInteger>(_ axis: [T]) -> Bool { ... }
}
```